tutorial/practicals/LFRic/building_code/Makefile

```diff
@@ -0,0 +1,49 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2020, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ------------------------------------------------------------------------------
+# Author: I. Kavcic, Met Office
+
+# Top-level Makefile to build all solutions of the LFRic hands-on tutorials on
+# building code.
+
+# Targets
+# build (default): complete build to run examples
+# test: PSyclone target for continuous integration
+# clean: Remove generated and compiled code
+.PHONY: build test clean
+
+build test clean:
+	${MAKE} -C 1_simple_kernels/solutions/part1 $@
+	${MAKE} -C 1_simple_kernels/solutions/part2 $@
+	${MAKE} -C 2_built_ins/solutions $@
```

Comment:
This may have been due to dependency on `KERNEL_OBJ` that was just copied and pasted from other Makefiles. I removed it for the built-ins tutorial. Please let me know if it works now.

Reply:
Ah, should I have been using the `test` target? If I do that then everything is fine!

Reply:
For me both `make test` and `make build` work for all solutions from this directory. Can you try `make build` from here again and see what happens? More importantly, do both targets from in the `2_built_ins/solutions` directory for you?

Reply:
Oh yes, `build` works OK. Maybe we should just change the default target to be either `test` or `build`? (`.DEFAULT_GOAL := build`)

Reply:
The default goal is build as it is listed as the first target and this happens when I run `make`. I agree it is good to explicitly specify it so I'll do this in the `Makefile`s.

Reply:
In fact, I changed `build` to `compile` to be in line with `examples`. Done for all `Makefile`s here.